sources/api/apiserver/src/server/mod.rs

```diff
@@ -354,6 +365,95 @@ async fn get_configuration_files(
     Ok(ConfigurationFilesResponse(resp))
 }
 
+/// Get the update status from 'thar-be-updates'
+async fn get_update_status() -> Result<UpdateStatusResponse> {
+    let output = Command::new("/usr/bin/thar-be-updates")
+        .arg("get-update-status")
+        .output()
+        .context(error::UpdateDispatcher)?;
+    if output.status.success() {
+        let resp: UpdateStatus =
+            serde_json::from_slice(&output.stdout).context(error::UpdateStatusParse {
+                stdout: output.stdout,
+            })?;
+        Ok(UpdateStatusResponse(resp))
+    } else {
+        let exit_status = match output.status.code() {
+            Some(code) => code,
+            None => output.status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(exit_status, Some(output.stderr)))
+    }
+}
+
+/// Refreshes the list of updates and checks if a 'version-lock'ed update is available
+async fn refresh_update() -> Result<HttpResponse> {
+    let status = Command::new("/usr/bin/thar-be-updates")
+        .arg("refresh-update")
+        .status()
```

Comment:
For reasons not entirely clear to me, I have to only wait for a exit status here. If I wait for both the status and process output, the spawned process will hang. 

This happens only if the spawned process forks a child process of its own. I suspect it has something to do with `Command::output()` stuck waiting for the process to finish writing to `stdout` and `stderr`. It does so by trying to read them until `EOF`. I guess if a child is forked, it duplicates the parent's file descriptors (including `stdout`, `stderr`) and keeps them open? And somehow that blocks the parent process from properly terminating for some reason. (Even though `thar-be-updates` would return immediately as a parent)

This is not a huge loss though, this just means we can't capture `stderr` here.

Reply:
>     * have `get_update_status()` read directly from the file (which should play nice with async)
> 
>     * have `thar-be-updates` take a flag to tell it what output file to manage

I'll expose the status file path from the `thar-be-updates` crate so `apiserver` knows where the status file is created. Then I'll update the helpers to read directly from the file. 


>     * ensure that we only use safe, atomic replacement for the status file
`thar-be-updates` should always be replacing the the status file atomically since write access is controlled by a file lock.

Reply:
> > ```
> > * ensure that we only use safe, atomic replacement for the status file
> > ```
> 
> `thar-be-updates` should always be replacing the the status file atomically since it's only replaced at the end of each command operation and write access is controlled by a file lock.

After discussing with @tjkirch offline, I'll go ahead and ensure this by first creating a tempfile and writing the status to it and making sure it succeeds before persisting it as the new status file.